File: Python/BOJ/5397.py
t = int(input())
for _ in range(t):
  string = list(input())
  left = []
  right = []
  for char in string:
    if char == "<":
      if left:
        right.append(left.pop())
    elif char == ">":
      if right:
        left.append(right.pop())
    elif char == "-":
      if left:
        left.pop()
    else:
      left.append(char)
  left += reversed(right)
  print("".join(left))

File: Swift/BOJ/문제집/연결리스트/5397.swift
//
//  5397.swift
//  Algorithm
//
//  Created by 유정주 on 2023/06/27.
//

import Foundation

let t = Int(readLine()!)!
for _ in 0..<t {
    let str = readLine()!
    var left = "", right = ""
    
    str.forEach {
        switch $0 {
        case "<":
            if !left.isEmpty {
                right.append(left.removeLast())
            }
        case ">":
            if !right.isEmpty {
                left.append(right.removeLast())
            }
        case "-":
            if !left.isEmpty {
                left.removeLast()
            }
        default:
            left.append(String($0))
        }
    }
    
    print(left + right.reversed())
}
